api/models/apd.js

```diff
@@ -186,6 +188,10 @@ const apdSchema = new mongoose.Schema({
   activities: [
     {
       _id: false,
+      activityId: {
+        type: String,
+        default: () => generateKey()
```

Comment:
it still has to happen on the front end because of some frustrating ways the front end is set up, I have the generate there as a default mostly for the seeds